docs/components/compact.md

```diff
@@ -53,27 +203,113 @@ Not setting this flag, or setting it to `0d`, i.e. `--retention.resolution-X=0d`
 
 Please note that blocks are only deleted after they completely "fall off" of the specified retention policy. In other words, the "max time" of a block needs to be older than the amount of time you had specified.
 
-## Storage space consumption
+## Deleting Aborted Partial Uploads
 
-In fact, downsampling doesn't save you any space but instead it adds 2 more blocks for each raw block which are only slightly smaller or relatively similar size to raw block. This is required by internal downsampling implementation which to be mathematically correct holds various aggregations. This means that downsampling can increase the size of your storage a bit (~3x), but it gives massive advantage on querying long ranges.
+It can happen that any producer started uploading some block, but never finished and never will. Sidecars will retry in case of failures during upload or process (unless there was no persistent storage),
+but very common case is with Compactor. If Compactor process crashes during upload of compacted block, whole compaction starts from scratch and new block ID is created. This means that
+partial upload will be never retried.
 
-## Groups
+To handle this case there is `--delete-delay=48h` flag that starts deletion of directories inside object storage without `meta.json` only after given time.
 
-The compactor groups blocks using the external_labels added by the Prometheus who produced the block.
-The labels must be both _unique_ and _persistent_ across different Prometheus instances.
+This value has to be smaller than upload duration and [consistency delay](#consistency-delay).
 
-By _unique_, we mean that the set of labels in a Prometheus instance must be different from all other sets of labels of
-your Prometheus instances, so that the compactor will be able to group blocks by Prometheus instance.
+## Halting
 
-By _persistent_, we mean that one Prometheus instance must keep the same labels if it restarts, so that the compactor will keep
-compacting blocks from an instance even when a Prometheus instance goes down for some time.
+Because of very specific nature of Compactor which is writing to object storage, potentially deleting sensitive data and downloading GBs of data, by default we halt Compator on certain data failures.
```

Comment:
```suggestion
Because of the very specific nature of Compactor which is writing to object storage, potentially deleting sensitive data, and downloading GBs of data, by default we halt Compactor on certain data failures.
```